### DISCUSSION TITLE
Getting weird error on different machine

### DISCUSSION THREAD

**Discussion by kaggrwal** [evrete/evrete#49] [Q&A] [answered]
Hi,

Everything is working fine on my system but when i deployed on another machine I'm getting the following error

`org.evrete.util.compiler.CompilationException: java.lang.NullPointerException: No Java compiler provided by this platform`.

Can anyone please help me with this, as i have already done the development with this and during test phase this error is coming. 

Java version on my system-  adopt@1.8.0-292

Java version on second machine- openjdk version "1.8.0_262"

**Comment by andbi** [comment, marked as answer]
Usually it happens when having both JDK and JRE installed and JRE is picked by your runtime environment. JREs do not have built-in Java compilers.

**Comment by kaggrwal** [comment]
Thanks @andbi , Let me check this on the machine

**Comment by kaggrwal** [comment]
This is because of JDK absence in the environment. Closing this, as it will be resolved by JDK
